extensions/amp-base-carousel/0.1/test-e2e/test-snap-property.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getSlides, getSpacers, prop} from './helpers';
+
+describes.endtoend(
+  'amp-base-carousel test snap property',
+  {
+    testUrl:
+      'http://localhost:8000/test/manual/amp-base-carousel/snap-property.amp.html',
+    environments: ['single'],
+  },
+  async function (env) {
+    let controller;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('should have snap property on spacers', async () => {
+      const slides = await getSlides(controller, '#snap-spacers');
+      const spacers = await getSpacers(controller);
+      for (let i = 0; i < slides.length; i++) {
+        const slide = slides[i];
+        const style = await prop(controller, slide, 'style');
+        const keys = Object.keys(style);
+        for (let j = 0; j < keys.length; j++) {
+          const key = keys[j];
+          await expect(style[key]).to.not.equal('scroll-snap-align');
+        }
+      }
```

Comment:
I'm a bit confused here. On the implementation you set the `scroll-snap-align` property:

```js
'scroll-snap-align': shouldSnap ? this.alignment_ : 'none',
```

If I understand correctly, this is checking `scroll-snap-align` as a style property value, not a property name:

```js
await expect(style[key]).to.not.equal('scroll-snap-align');
```

Shouldn't this be something like...?

```suggestion
      for (let i = 0; i < slides.length; i++) {
        const slide = slides[i];
        const style = await prop(controller, slide, 'style');
        expect(style.scrollSnapAlign).to.equal('none');
      }
```

Reply:
Discussed offline: for e2e tests, using `getElementProperty` to get the element.style returns an Array.